Guix build w/o substitutes failing at libgit2 check

This is an upstream Guix issue ([55273](https://debbugs.gnu.org/cgi/bugreport.cgi?bug=55273)), but they seem to be planning to resolve it with a version bump, and either way we have versions pinned I think.

```
building /gnu/store/myai2xf4f0z2nr331mx2y4xr4jv1lwid-libgit2-1.3.0.drv...
| 'unpack' phaseild-log 356 178
`/gnu/store/xa8yg9nvawhfvk6hv8lhif9z5pybzan5-libgit2-1.3.0-checkout/tests/resources/status/staged_delete_modified_file' -> `./tests/resources/status/staged_delete_modified_file'
 17% [########################                                                                                                                           ]ild-log 356 189
/tmp/guix-build-libgit2-1.3.0.drv-0/source/src/regexp.c:148:5: warning: ?error? may be used uninitialized in this function [-Wmaybe-uninitialized]
  148 |  if (error < 0)
      |     ^
 17% [########################                                                                                                                           ]ild-log 356 72
[ 18%] Building C object src/CMakeFiles/git2internal.dir/repository.c.o
| 'check' phasenote: keeping build directory `/tmp/guix-build-libgit2-1.3.0.drv-0'
builder for `/gnu/store/myai2xf4f0z2nr331mx2y4xr4jv1lwid-libgit2-1.3.0.drv' failed with exit code 1
build of /gnu/store/myai2xf4f0z2nr331mx2y4xr4jv1lwid-libgit2-1.3.0.drv failed
```

```
refs::revparse.................F

  1) Failure:
refs::revparse::date [/tmp/guix-build-libgit2-1.3.0.drv-0/source/tests/refs/revparse.c:31]
  Function call succeeded: error
  no error, expected non-zero return
```

--- a/contrib/guix/libexec/prelude.bash
+++ b/contrib/guix/libexec/prelude.bash
@@ -51,7 +51,7 @@ fi
 time-machine() {
     # shellcheck disable=SC2086
     guix time-machine --url=https://git.savannah.gnu.org/git/guix.git \
-                      --commit=34e9eae68c9583acce5abc4100add3d88932a5ae \
+                      --commit=998eda3067c7d21e0d9bb3310d2f5a14b8f1c681 \
                       --cores="$JOBS" \
                       --keep-failed \
                       --fallback \

